pymc3/distributions/distribution.py

```diff
@@ -232,99 +232,111 @@ def draw_values(params, point=None, size=None):
-    # parents and children to replace them.
-    leaf_nodes = {}
-    named_nodes_parents = {}
-    named_nodes_children = {}
-    for param in params:
-        if hasattr(param, 'name'):
-            # Get the named nodes under the `param` node
-            nn, nnp, nnc = get_named_nodes_and_relations(param)
-            leaf_nodes.update(nn)
-            # Update the discovered parental relationships
-            for k in nnp.keys():
-                if k not in named_nodes_parents.keys():
-                    named_nodes_parents[k] = nnp[k]
-                else:
-                    named_nodes_parents[k].update(nnp[k])
-            # Update the discovered child relationships
-            for k in nnc.keys():
-                if k not in named_nodes_children.keys():
-                    named_nodes_children[k] = nnc[k]
-                else:
-                    named_nodes_children[k].update(nnc[k])
-
-    # Init givens and the stack of nodes to try to `_draw_value` from
-    givens = {}
-    stored = set()  # Some nodes
-    stack = list(leaf_nodes.values())  # A queue would be more appropriate
-    while stack:
-        next_ = stack.pop(0)
-        if next_ in stored:
-            # If the node already has a givens value, skip it
-            continue
-        elif isinstance(next_, (tt.TensorConstant,
-                                tt.sharedvar.SharedVariable)):
-            # If the node is a theano.tensor.TensorConstant or a
-            # theano.tensor.sharedvar.SharedVariable, its value will be
-            # available automatically in _compile_theano_function so
-            # we can skip it. Furthermore, if this node was treated as a
-            # TensorVariable that should be compiled by theano in
-            # _compile_theano_function, it would raise a `TypeError:
-            # ('Constants not allowed in param list', ...)` for
-            # TensorConstant, and a `TypeError: Cannot use a shared
-            # variable (...) as explicit input` for SharedVariable.
-            stored.add(next_.name)
-            continue
+    # Get the nodes that we must draw from as list of lists
+    try:
+        model = modelcontext(model)
+        dependence_dag, index = get_sub_dag(model.dependence_dag,
+                                            params,
+                                            return_index=True)
+    except Exception:
+        dependence_dag, index = get_sub_dag(DiGraph(),
+                                            params,
+                                            return_index=True)
+    # Store list of nodes in reversed topological sort order, i.e. the nodes
+    # without ancestors at the end. This way, default pop() will get the
+    # correct node, and append() will add a node for the next iteration
+    queue = list(reversed(list(topological_sort(dependence_dag))))
+
+    # Init drawn values and updatable point and givens
+    drawn = {n: None for n in dependence_dag.nodes}
```

Comment:
I don't think you need to initialize the names here as you just below check if the key is None. Thus you should be able to replace this with `drawn = {}` and then below  `if node in drawn:`